app/models/user.rb

```diff
@@ -16,6 +16,26 @@ class User < ActiveRecord::Base
   has_many :magics
   has_many :universes
 
+  # as_json would try to print the password digest, which requires authentication
+  def as_json(options={})
+    excludes = [:password_digest, :old_password]
+    options = {} if options.nil?
+    options[:except] ||= excludes
+    super(options)
+  end
+
+  def to_json(options={})
+    options[:except] ||= [:password_digest, :old_password]
```

Comment:
We probably want to define this list on the model somewhere so we can reuse it and modify it in a single place, instead of hard-coding it in these handful of places.